From ea52b42e86c5f168a58a9130a21c8d2654b8ee8f Mon Sep 17 00:00:00 2001
From: Valentin Caron <valentin.caron@foss.st.com>
Date: Mon, 13 May 2024 11:53:28 +0200
Subject: [PATCH 0883/1695] plat-stm32mp2: scmi_server: enable OPTEE_SMT module
 in SCPFW

Enable compilation of OPTEE_SMT module in SCPFW.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git fc6f85765ab9f3479420adcbe43bb4fd02a2543b plat-stm32mp2: scmi_server: enable OPTEE_SMT module in SCPFW]
Co-developed-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Signed-off-by: Raphael Gallais-Pou <raphael.gallais-pou@foss.st.com>
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I7d1ff0ff0d0505398af5eeb0d46c2ba1a1a03759
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/380387
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/lib/scmi-server/conf-optee-stm32mp2.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/lib/scmi-server/conf-optee-stm32mp2.mk b/core/lib/scmi-server/conf-optee-stm32mp2.mk
index 76bfeaeec..471e3f049 100644
--- a/core/lib/scmi-server/conf-optee-stm32mp2.mk
+++ b/core/lib/scmi-server/conf-optee-stm32mp2.mk
@@ -5,6 +5,9 @@ $(call force,CFG_SCPFW_MOD_OPTEE_CLOCK,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_CONSOLE,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_MBX,y)
 $(call force,CFG_SCPFW_MOD_OPTEE_RESET,y)
+ifeq ($(CFG_SCMI_CORTEXM_AGENT),y)
+$(call force,CFG_SCPFW_MOD_OPTEE_SMT,y)
+endif
 $(call force,CFG_SCPFW_MOD_OPTEE_VOLTD_REGULATOR,y)
 $(call force,CFG_SCPFW_MOD_PSU,y)
 $(call force,CFG_SCPFW_MOD_PSU_OPTEE_REGULATOR,y)
-- 
2.47.2

